sklearn/utils/metaestimators.py

```diff
@@ -104,11 +105,13 @@ def __get__(self, obj, owner=None):
             # delegate only on instances, not the classes.
             # this is to allow access to the docstrings.
             if not self.check(obj):
-                raise AttributeError(
-                    f"This {repr(owner.__name__)}"
-                    " has no attribute"
+                msg = (
+                    f"This {repr(owner.__name__)} has no attribute"
                     f" {repr(self.attribute_name)}"
+                    if self.err_msg is None
+                    else self.err_msg
                 )
+                raise AttributeError(msg)
```

Comment:
I prefer do no change here, don't need add additional err_msg args, but add doc say allow `check` to return True/False or raise error.

then the Available_if descriptor will also raise the original error if attribute not available.

Reply:
> I prefer do no change here, don't need add additional err_msg args

Could you elaborate on this?

Reply:
add a new argument, then the mlflow patch need extra patch code, for fitting different version `@available_if`

Reply:
> I think err_msg allows us to show a more helpful error message. Just saying attribute xxx is not available might not be helpful for users.

We can allow `check` raise AssertError with any message. So this is not an issue.